test/io_test.dart

```diff
@@ -344,6 +348,59 @@ void testExistencePredicate(String name, bool Function(String path) predicate,
       });
     }
   });
+  group('extractTarGz', () {
+    test('decompresses simple archive', () async {
+      await withTempDir((tempDir) async {
+        await extractTarGz(
```

Comment:
Can we make a test case that we default to `skip: true` where we iterate through all packages and all versions on `pub.dev` and check if the tar archive that is downloaded can be extracted?

Then we just manually enable or run the test when we want to run it. Maybe we can create a test tag called `extremely_expensive` and [tag the test case](https://pub.dev/packages/test#tagging-tests) with this and configuring the tag to be skipped by default.

Reply:
Sounds good.
I'll look into that.

Reply:
I ended up putting it in tools/ instead. It is hard to run in a single session, so I made it save a status file, so you can pick up from where you left off.